octobot_trading/personal_data/orders/order.py

```diff
@@ -398,13 +426,105 @@ def get_position_side(self, future_contract):
             return enums.PositionSide.BOTH
         raise errors.InvalidPositionSide(f"Can't determine order position side while using hedge position mode")
 
+    @contextlib.contextmanager
+    def active_or_inactive_transition(self):
+        previous_value = self.is_in_active_inactive_transition
+        try:
+            self.is_in_active_inactive_transition = True
+            yield
+        finally:
+            self.is_in_active_inactive_transition = previous_value
+
+    def create_or_update_active_trigger(self, active_trigger_price: decimal.Decimal, active_trigger_above: bool):
+        if self.active_trigger is None:
+            self.active_trigger = price_trigger.PriceTrigger(
+                self.on_active_trigger, (None, None), active_trigger_price, active_trigger_above
+            )
+        else:
+            self.active_trigger.trigger_price = active_trigger_price
+            self.active_trigger.trigger_above = active_trigger_above
+
+    async def set_as_inactive(self, active_trigger_price: decimal.Decimal, active_trigger_above: bool):
+        """
+        Marks the instance as inactive and ensures the inactive order watcher is scheduled.
+        """
+        if active_trigger_price is None or active_trigger_above is None:
+            raise ValueError(
+                f"Both active_trigger_price and active_trigger_above must be provided to set an order as inactive"
+            )
+        logging.get_logger(self.get_logger_name()).info("Order is switching to inactive")
+        self.is_active = False
+        self.create_or_update_active_trigger(active_trigger_price, active_trigger_above)
+        # enforce attributes in case order has been canceled
+        self.status = enums.OrderStatus.OPEN
+        self.canceled_time = 0
+        await self._ensure_inactive_order_watcher()
```

Comment:
I don't think this is consistent with the trigger architecture. I believe Order should only be able to add/update a trigger but not create it. In other words, it should not take an 'active_trigger_price' argument but only an 'active_trigger,' and it’s the responsibility of the caller to provide the instance of the correct trigger class based on what is desired.

Reply:
good point, i'll update this, thanks !

Reply:
it's up